File: apps/dot-net-sdk/Firecrawl/Models/VideoItem.cs
using System.Text.Json.Serialization;

namespace Firecrawl.Models;

/// <summary>
/// Represents a discovered video reference on a scraped page.
/// </summary>
public class VideoItem
{
    [JsonPropertyName("url")]
    public string? Url { get; set; }

    [JsonPropertyName("sourceURL")]
    public string? SourceURL { get; set; }

    [JsonPropertyName("source")]
    public string? Source { get; set; }

    [JsonPropertyName("kind")]
    public string? Kind { get; set; }

    [JsonPropertyName("provider")]
    public string? Provider { get; set; }

    [JsonPropertyName("title")]
    public string? Title { get; set; }

    [JsonPropertyName("thumbnail")]
    public string? Thumbnail { get; set; }

    [JsonPropertyName("description")]
    public string? Description { get; set; }

    [JsonPropertyName("duration")]
    public string? Duration { get; set; }

    [JsonPropertyName("mimeType")]
    public string? MimeType { get; set; }

    [JsonPropertyName("width")]
    public int? Width { get; set; }

    [JsonPropertyName("height")]
    public int? Height { get; set; }

    [JsonPropertyName("metadata")]
    public Dictionary<string, object>? Metadata { get; set; }
}

File: apps/java-sdk/src/main/java/com/firecrawl/models/VideoItem.java
package com.firecrawl.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.Map;

/**
 * A discovered video reference on a scraped page.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class VideoItem {
    private String url;

    @JsonProperty("sourceURL")
    private String sourceURL;

    private String source;
    private String kind;
    private String provider;
    private String title;
    private String thumbnail;
    private String description;
    private String duration;
    private String mimeType;
    private Integer width;
    private Integer height;
    private Map<String, Object> metadata;

    public String getUrl() { return url; }
    public String getSourceURL() { return sourceURL; }
    public String getSource() { return source; }
    public String getKind() { return kind; }
    public String getProvider() { return provider; }
    public String getTitle() { return title; }
    public String getThumbnail() { return thumbnail; }
    public String getDescription() { return description; }
    public String getDuration() { return duration; }
    public String getMimeType() { return mimeType; }
    public Integer getWidth() { return width; }
    public Integer getHeight() { return height; }
    public Map<String, Object> getMetadata() { return metadata; }
}
